File: Front/tpi-teg-grupo-11/src/app/core/models/class/player.ts
import {User} from './user';
import {Objective} from './objective';
import {Color} from '../../enums/color';
import {PlayerType} from '../../enums/player-type';
import {TerritoryState} from './game-data';

export interface Player {
  playerId?: number;
  user?: User;
  gameId: string;
  objective?: Objective;
  playerColor?: Color;
  turnOrder: number;
  isWinner: boolean;
  hasLost: boolean;
  difficulty?: string;
  playerType: PlayerType;
  territories?: TerritoryState[];
}

File: Back/tpi-teg-grupo-11/src/main/java/ar/edu/utn/frc/tup/piii/models/Player.java
package ar.edu.utn.frc.tup.piii.models;

import ar.edu.utn.frc.tup.piii.entities.CardStateEntity;
import ar.edu.utn.frc.tup.piii.enums.BotDifficulty;
import ar.edu.utn.frc.tup.piii.enums.Color;
import ar.edu.utn.frc.tup.piii.enums.PlayerType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Player {
    private Long playerId;
    private UUID gameId;
    private User user;
    private Objective objective;
    private Color playerColor;
    private int turnOrder;
    private boolean isWinner;
    private boolean hasLost;
    private BotDifficulty difficulty;
    private PlayerType playerType;
    private List<CardStateEntity> cards;
//    private List<GlobalPactEntity> globalPactsStarted;
//    private List<GlobalPactEntity> globalPactsAccepted;
    private List<TerritoryState> territories;
}